src/change_stream.ts

```diff
@@ -648,21 +648,25 @@ export class ChangeStream<
   hasNext(callback?: Callback): Promise<boolean> | void {
     this._setIsIterator();
     return maybeCallback(async () => {
-      try {
-        const hasNext = await this.cursor.hasNext();
-        return hasNext;
-      } catch (error) {
+      // Change streams must resume indefinitely as long while each resume event succeeds.
```

Comment:
Not sure the best way to fix "as long while", but it doesn't work semantically at the moment